src/libcephsqlite.cc

```diff
@@ -0,0 +1,927 @@
+#include <unistd.h>
+#include <string.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <limits.h>
+
+
+#include "include/rados/librados.hpp"
+#include "include/radosstriper/libradosstriper.hpp"
+#include "common/ceph_mutex.h"
+
+static int _cephsqlite3_Close(sqlite3_file *pFile);
+static int _cephsqlite3_Read(sqlite3_file *pFile, void *zBuf, int iAmt, sqlite_int64 iOfst);
+static int _cephsqlite3_Write(sqlite3_file *pFile, const void *zBuf, int iAmt, sqlite_int64 iOfst);
+static int _cephsqlite3_Truncate(sqlite3_file *pFile, sqlite_int64 size);
+static int _cephsqlite3_Sync(sqlite3_file *pFile, int flags);
+static int _cephsqlite3_FileSize(sqlite3_file *pFile, sqlite_int64 *pSize);
+static int _cephsqlite3_Lock(sqlite3_file *pFile, int eLock);
+static int _cephsqlite3_Unlock(sqlite3_file *pFile, int eLock);
+static int _cephsqlite3_CheckReservedLock(sqlite3_file *pFile, int *pResOut);
+static int _cephsqlite3_FileControl(sqlite3_file *pFile, int op, void *pArg);
+static int _cephsqlite3_SectorSize(sqlite3_file *pFile);
+static int _cephsqlite3_DeviceCharacteristics(sqlite3_file *pFile);
+
+static int _cephsqlite3_Open(sqlite3_vfs *pVfs, const char *zName, sqlite3_file *pFile, int flags, int *pOutFlags);
+static int _cephsqlite3_Delete(sqlite3_vfs *pVfs, const char *zPath, int dirSync);
+static int _cephsqlite3_Access(sqlite3_vfs *pVfs, const char *zPath, int flags, int *pResOut);
+static int _cephsqlite3_FullPathname(sqlite3_vfs *pVfs, const char *zPath, int nPathOut, char *zPathOut);
+static void *_cephsqlite3_DlOpen(sqlite3_vfs *pVfs, const char *zPath);
+static void _cephsqlite3_DlError(sqlite3_vfs *pVfs, int nByte, char *zErrMsg);
+static void (*_cephsqlite3_DlSym(sqlite3_vfs *pVfs, void *pH, const char *z))(void);
+static void _cephsqlite3_DlClose(sqlite3_vfs *pVfs, void *pHandle);
+static int _cephsqlite3_Randomness(sqlite3_vfs *pVfs, int nByte, char *zByte);
+static int _cephsqlite3_Sleep(sqlite3_vfs *pVfs, int nMicro);
+static int _cephsqlite3_CurrentTime(sqlite3_vfs *pVfs, double *pTime);
+
+static sqlite3_vfs *_cephsqlite3__vfs(void);
+static librados::IoCtx *get_io_ctx(const char *zPath);
+static libradosstriper::RadosStriper *get_radosstriper(const char *zPath);
+static std::string get_db_name(const char *zName);
+static int get_lock_type(const std::string &db_name, const std::string &file_name);
+static void set_lock_type(const std::string &db_name, const std::string &file_name, int eLock);
+static std::string get_lock_file_name(const std::string &db_name, int eLock);
+static int create_lock_files(librados::IoCtx *io_ctx, const std::string &db_name, bool must_create);
+static int remove_lock_files(librados::IoCtx *io_ctx, const std::string &db_file);
+static int lock_file_in_rados(librados::IoCtx *io_ctx, const std::string &lock_file_name, const std::string &lock_name, bool exclusive);
+
+
+struct CephVFSContext {
+  librados::IoCtx               *io_ctx = nullptr;
```

Comment:
i'd suggest remove the extra spaces for the alignment to be consistent with the rest of the code base.